rust/libs/logging/src/windows_event_log.rs

```diff
@@ -0,0 +1,456 @@
+    }
+
+    for (key, value) in fields {
+        if !output.is_empty() {
+            output.push('\n');
+        }
+        let _ = write!(output, "{key}: {value}");
+    }
+
+    output
+}
+
+fn try_register_source(source: &str) -> Result<(), Error> {
+    let key_path = format!("SYSTEM\\CurrentControlSet\\Services\\EventLog\\Application\\{source}");
+    let key_path_wide = to_wide_string(&key_path);
+    let mut hkey = windows::Win32::System::Registry::HKEY::default();
+
+    unsafe {
+        RegCreateKeyExW(
+            HKEY_LOCAL_MACHINE,
+            PCWSTR(key_path_wide.as_ptr()),
+            Some(0),
+            PCWSTR::null(),
+            REG_OPTION_NON_VOLATILE,
+            KEY_WRITE,
+            None,
+            &mut hkey,
+            None,
+        )
+        .ok()
+        .map_err(|e| Error::CreateRegistryKey { error: e })?;
+
+        let result = (|| {
+            RegSetValueExW(
+                hkey,
+                w!("TypesSupported"),
+                Some(0),
+                REG_DWORD,
+                Some(TYPES_SUPPORTED.to_le_bytes().as_slice()),
+            )
+            .ok()
+            .map_err(|e| Error::SetRegistryValue { error: e })?;
+
+            let message_file = "%SystemRoot%\\System32\\EventCreate.exe";
+            let message_file_wide = to_wide_string(message_file);
+            let message_file_bytes: Vec<u8> = message_file_wide
+                .iter()
+                .flat_map(|&word| word.to_le_bytes())
+                .collect();
+
+            RegSetValueExW(
+                hkey,
+                w!("EventMessageFile"),
+                Some(0),
+                REG_EXPAND_SZ,
+                Some(&message_file_bytes),
+            )
+            .ok()
+            .map_err(|e| Error::SetRegistryValue { error: e })?;
+
+            Ok(())
+        })();
+
+        let _ = RegCloseKey(hkey);
+        result
+    }
+}
+
+fn to_wide_string(s: &str) -> Vec<u16> {
+    s.encode_utf16().chain(std::iter::once(0)).collect()
+}
+
+/// Creates a Windows Event Log layer with filtering from `EVENTLOG_DIRECTIVES` (default: `info`).
+pub fn layer<S>(source: &str) -> Result<Filtered<WindowsEventLogLayer, EnvFilter, S>, Error>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    let directives =
+        std::env::var(ENV_EVENTLOG_DIRECTIVES).unwrap_or_else(|_| DEFAULT_DIRECTIVES.to_owned());
+    layer_with_directives(source, &directives)
+}
+
+/// Creates a Windows Event Log layer with custom filter directives.
+pub fn layer_with_directives<S>(
+    source: &str,
+    directives: &str,
+) -> Result<Filtered<WindowsEventLogLayer, EnvFilter, S>, Error>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    let layer = WindowsEventLogLayer::new(source)?;
+    let filter = EnvFilter::try_new(directives).map_err(|error| Error::InvalidDirectives {
+        directives: directives.to_owned(),
+        error: error.to_string(),
+    })?;
+    Ok(layer.with_filter(filter))
+}
```

Comment:
I think traditionally, these tracing layers only do one thing and you let the user compose things on top of it. I'd suggest we remove the `EnvFilter` here and make this a concern of the caller, if and how they want to filter.